pkg/backup/compaction_job.go

```diff
@@ -523,17 +523,46 @@ func updateCompactionBackupDetails(
 	return compactedDetails, nil
 }
 
-// compactIntroducedSpans takes a compacted backup manifest and the full chain of backups it belongs
-// to and computes the introduced spans for the compacted backup.
-func compactIntroducedSpans(
-	ctx context.Context, manifest backuppb.BackupManifest, chain compactionChain,
-) (roachpb.Spans, error) {
-	if err := checkCoverage(ctx, manifest.Spans, chain.backupChain); err != nil {
+// createCompactionManifest creates a new manifest for a compaction job and its
+// compacted chain. The details should have its targets resolved.
+func (c compactionChain) createCompactionManifest(
+	ctx context.Context, details jobspb.BackupDetails,
+) (*backuppb.BackupManifest, error) {
+	// TODO (kev-cao): Will need to update the SSTSinkKeyWriter to support
+	// range keys.
+	lastBackup := c.lastBackup()
+	if len(lastBackup.Tenants) != 0 {
+		return nil, errors.New("backup compactions does not yet support range keys")
+	}
+	introducedSpans, err := c.compactIntroducedSpans(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cManifest := lastBackup
+	cManifest.StartTime = c.start
+	cManifest.EndTime = c.end
+	cManifest.Descriptors = details.ResolvedTargets
+	cManifest.CompleteDbs = details.ResolvedCompleteDbs
+	cManifest.IntroducedSpans = introducedSpans
+	cManifest.Files = nil
+	cManifest.DescriptorChanges = nil // Revision history is not supported.
```

Comment:
i think we can remove this line and the ParitionDescriptorFileNames line. we already check that we cannot compact off of a chain with these.

Reply:
`DescriptorChanges` was niled out because otherwise we get a `assertion: this placeholder Descriptor entry should never be used`.  Although `PartitionDescriptorFileNames` doesn't need to be niled out though, I mostly just added that for completion since I was already niling out `DescriptorChanges`